lib/components/user/monitored-trip/trip-notifications-pane.js

```diff
@@ -72,6 +68,66 @@ const Select = ({ children, Control = FormControl, label, name }) => (
   </>
 )
 
+function Options ({ defaultValue, options }) {
+  // <FormattedMessage> can't be used inside <option>.
+  const intl = useIntl()
+  return options.map(({ text, value }, i) => (
+    <option key={i} value={value}>
+      {value === defaultValue
+        ? intl.formatMessage(
+          { id: 'common.forms.defaultValue' },
+          { value: text }
+        )
+        : text
+      }
+    </option>
+  ))
+}
+
+const basicYesNoOptions = [
+  {
+    id: 'yes',
+    value: true
+  },
+  {
+    id: 'no',
+    value: false
+  }
+]
+
+/**
+ * Produces a yes/no list of options with the specified
+ * default value (true for yes, false for no).
+ */
+function YesNoOptions ({ defaultValue }) {
+  // <FormattedMessage> can't be used inside <option>.
+  const intl = useIntl()
+  const options = basicYesNoOptions.map(({ id, value }) => ({
+    text: intl.formatMessage({id: `common.forms.${id}`}),
+    value
+  }))
+  return <Options defaultValue={defaultValue} options={options} />
+}
+
+/**
+ * Produces a list of duration options with the specified default value.
+ */
+function DurationOptions ({ defaultValue, minuteOptions }) {
+  // <FormattedMessage> can't be used inside <option>.
+  const intl = useIntl()
+  const options = minuteOptions.map(minutes => ({
+    text: intl.formatMessage(
+      { id: minutes === 60
+        ? 'components.TripNotificationsPane.oneHour'
+        : 'common.time.tripDurationFormatZeroHours'
+      },
```

Comment:
Please do not create dynamic message IDs as it is hard to automatically keep track of which IDs are in the code. Instead, please create a separate variable for each possible message.